paasta_tools/paasta_metastatus.py

```diff
@@ -63,7 +64,30 @@ def get_mesos_cpu_status(metrics):
     return total, used, available
 
 
+def combine_key_value_pair_iterators(iterator1, iterator2, collision_operator, valid_keys=None):
```

Comment:
I'm a bit skeptical about these kin of functions, which kinda hint that we may not be returning manageable datastructures back in the first place. Lets talk about this in person.